### PR TITLE
Fix for Jira Issue - EMM-967

### PR DESCRIPTION
**Pull request by Kamidu**

### CODE REVIEW

File: modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js [hunk 1]
```diff
@@ -1,3 +1,37 @@
...
+
+function post(path, params, method) {
+    method = method || "post"; // Set method to post by default if not specified.
+
+    // The rest of this code assumes you are not using a library.
+    // It can be made less wordy if you use one.
+    var form = document.createElement("form");
+    form.setAttribute("method", method);
+    form.setAttribute("action", path);
+
```
Comment (harshanL) [inline modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js]: Check formatting of if and for blocks.

File: modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js [hunk 2]
```diff
@@ -1,3 +1,37 @@
...
+function utf8_to_b64( str ) {
```
Comment (kasundananjaya) [inline modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js]: If these kind of methods are re-usable, can we move them to a util JS?

File: modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js [hunk 3]
```diff
@@ -1,3 +1,37 @@
...
+    return window.btoa(unescape(encodeURIComponent( str )));
+}
+
+function b64_to_utf8( str ) {
```
Comment (kasundananjaya) [inline modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js]: If these kind of methods are re-usable, can we move them to a util JS?

Comment (Kamidu) [inline modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js]: uill.js is a backend js file while this is a front end js file.
for increase re-usability of methods which requires from various units we can put those front js methods into a single unit and include that in every unit which requires those functionalities needed.WDYT?

File: modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js (hunk 2 above)
Comment (Kamidu) [inline modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js]: uill.js is a backend js file while this is a front end js file.
for increase re-usability of methods which requires from various units we can put those front js methods into a single unit and include that in every unit which requires those functionalities needed.WDYT?

File: modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js (hunk 1 above)
Comment (Kamidu) [inline modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js]: These formatting issues has been fixed in a recent PR

File: modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js [hunk 4]
```diff
@@ -1,3 +1,37 @@
...
+    method = method || "post"; // Set method to post by default if not specified.
+
+    // The rest of this code assumes you are not using a library.
+    // It can be made less wordy if you use one.
+    var form = document.createElement("form");
+    form.setAttribute("method", method);
+    form.setAttribute("action", path);
+
+    for(var key in params) {
+        if(params.hasOwnProperty(key)) {
```
Comment (prabathabey) [inline modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js]: Fir formatting issues.

Comment (Kamidu) [inline modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js]: These formatting issues has been fixed in a recent PR

File: modules/apps/jaggery/mdm/src/mdm/units/login/login.hbs [hunk 5]
```diff
@@ -8,7 +8,7 @@
         <span class="wr-validation-summary hidden">
                 <p>Login failed. Check your credentials and try again.</p>
         </span>
-        <form method="POST" class="form-login-box" action="{{loginPath}}">
+        <form id=login method="POST" class="form-login-box" action="{{loginPath}}" >
```
Comment (dakshika) [inline modules/apps/jaggery/mdm/src/mdm/units/login/login.hbs]: unquoted attributes are valid on html5 specification, but as a best practice do recommend to use quotes.

File: modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js [hunk 6]
```diff
@@ -1,3 +1,37 @@
+
+function utf8_to_b64( str ) {
+    return window.btoa(unescape(encodeURIComponent( str )));
```
Comment (dakshika) [inline modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js]: this is a know function for b64 conversion, but both unescape and escape are  deprecated. try to use latest

File: modules/apps/jaggery/mdm/src/mdm/units/login/login.hbs (hunk 5 above)
Comment (Kamidu) [inline modules/apps/jaggery/mdm/src/mdm/units/login/login.hbs]: Duly Noted and thank you for the recommendation.
Recommended fixes can be found in PR[1]

[1]. #166

File: modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js (hunk 6 above)
Comment (Kamidu) [inline modules/apps/jaggery/mdm/src/mdm/units/login/public/js/login-box.js]: There is a discussion goin on regarding this on dev mailing list under the subject "[Dev][Encoding Special Characters] Throws an Character Out Of Range exception" .
An Update will be provided accordingly to the outcome of the discussion.

**Pull request merged**